docs/snapshotting/snapshot-support.md

```diff
@@ -0,0 +1,292 @@
+
+If a `version` is specified, the new snapshot is saved at that version, otherwise
+it will be saved at the same version of the running Firecracker. The version is only
+used for the microVM state file as it contains internal state structures for device
+emulation, vCPUs and others that can change their format from a Firecracker version
+to another. Versioning is not required for the block and memory files. The separate
+block device file components of the snapshot have to be handled by the user.
+
+- _on failure_: no side-effects.
+
+### Creating diff snapshots
+
+For creating a diff snapshot, you should use the same API command, but with
+`snapshot_type` field set to `Diff`.
+*Note*: If not specified, `snapshot_type` is by default `Full`.
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PUT "http://localhost/snapshot/create" \
+    -H  "accept: application/json" \
+    -H  "Content-Type: application/json" \
+    -d "{
+            \"snapshot_type\": \"Diff\",
+            \"snapshot_path\": \"/path/to/snapshot_file\",
+            \"mem_file_path\": \"/path/to/mem_file\",
+            \"version\": \"2\"
+    }"
+```
+
+**Prerequisites**: The microVM is `Paused`.
+                   On a fresh microVM, `track_dirty_pages` field should be set to `true`,
+                   when configuring the `/machine-config` resource, while on a snapshot
+                   loaded microVM, `enable_diff_snapshots` from `PUT /snapshot/load`
+                   request body, should be set.
+                   
+**Effects**:
+- _on success_:
+The file indicated by `/path/to/snapshot_file` contains the devices' model state and
+emulation state, same as when creating a full snapshot. The one indicated by
+`/path/to/mem_file` contains this time a **diff copy** of the guest memory - the diff
+consists of the memory pages which have been dirtied since the last snapshot creation
+or since the creation of the microVM, whichever of these events was the most recent.
+All the other effects mentioned in the **Effects** section from `## Creating full snapshots`
+apply here.
+- _on failure_: no side-effects.
+
+*Note*: This is an example of an API command that enables dirty page tracking:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i  \
+    -X PUT 'http://localhost/machine-config' \
+    -H 'Accept: application/json'            \
+    -H 'Content-Type: application/json'      \
+    -d '{
+            "vcpu_count": 2,
+            "mem_size_mib": 1024,
+            "ht_enabled": false,
+            "track_dirty_pages": true
+    }'
+```
+
+Enabling this support enables KVM dirty page tracking, so it comes at a cost
+(which consists of CPU cycles spent by KVM accounting for dirtied pages); it should only
+be used when needed.
+
+Creating a snapshot will **not** influence state, will **not** stop or end the microVM,
+it can be used as before, so the microVM can be resumed if you still want to use it.
```

Comment:
nit: I would stress the disks story even further here:
```
Users should make sure to copy disk backing files at this point if they plan to continue using the current microVM.
```

Reply:
Done.